src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -80,6 +81,8 @@ public ProduceService(Map<String, Object> props, String name) throws Exception {
     _brokerList = config.getString(ProduceServiceConfig.BOOTSTRAP_SERVERS_CONFIG);
     String producerClass = config.getString(ProduceServiceConfig.PRODUCER_CLASS_CONFIG);
 
+    String partitionerClassName = config.getString(ProduceServiceConfig.PARTITIONER_CLASS_CONFIG);
+    _partitioner = (Partitioner) Class.forName(partitionerClassName).getConstructor().newInstance();
```

Comment:
I am not sure. But is it possible to just do `config.getConfiguredInstance(ProduceServiceConfig.PARTITIONER_CLASS_CONFIG, Partitioner)`?